### PR TITLE
Refactoring filtered replication && add device function

### PR DESCRIPTION
**Pull request by kosssi**

### CODE REVIEW

File: doc/filtered_replication.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * a password for authentication: `********`
+ * a configuration
+ * a callback
+
+```javascript
+config = {file: true, folder: true}
+cozyUrl = 'https://demo.cozycloud.cc'
+deviceName = 'demo-laptop'
+password = '********'
+filteredReplication.setFilteredDoc(cozyUrl, deviceName, password, config, callback);
```
Comment (nono) [inline doc/filtered_replication.md]: Maybe we can call this function `setDesignDoc`

File: package.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "cozy-device-sdk",
-  "version": "1.0.0",
+  "version": "1.2.0",
```
Comment (nono) [inline package.json]: Why no `1.1.0` version?

Comment (kosssi) [inline package.json]: 1.1.0 : refactoring
1.2.0 : add device function 

:smile:

File: doc/filtered_replication.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Configuration
+-------------
+
+The configuration allow you what you want synchronize:
+
+```javascript
+config = {
+    file: true,
+    folder: true,
```
Comment (nono) [inline doc/filtered_replication.md]: I think we should do the same for file that for calendar, ie get the associated docTypes. With, `calendar: true`, the docTypes `event` and `tag` are kept. With `file: true`, I think we should keep `file` and `folder` docTypes. What do you think?

File: doc/filtered_replication.md (hunk 1 above)
Comment (kosssi) [inline doc/filtered_replication.md]: :+1:

File: src/filtered_replication.coffee [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+module.exports =
+
+
+    getFilterName: (deviceName) ->
+        log.debug "getFilterName"
+
+        "filter-#{deviceName}-config"
+
+
+    getFilteredDocId: (deviceName) ->
```
Comment (nono) [inline src/filtered_replication.coffee]: I'd call it `getDesignDocId`

File: test/unit/filtered_replication.coffee [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+            doc._id.should.be.equal id
+            doc.views.should.be.empty()
+            doc.filters.should.be.eql "#{key}": value
+
+    describe "filtered function", ->
+
+        it "create empty filter function.", ->
+
+            fn = eval('(' + filteredReplication.getFilteredFunction() + ')')
```
Comment (nono) [inline test/unit/filtered_replication.coffee]: You could use string interpolation as below: https://github.com/cozy/cozy-device-sdk/pull/5/files#diff-919da6254b54bfa4c3ee3795c6f4e982R59

File: doc/filtered_replication.md (hunk 3 above)
Comment (kosssi) [inline doc/filtered_replication.md]: :bulb: it's good idea.

**Comment by aenario** [comment]
Does the autopublish works ?

**Comment by aenario** [comment]
Cause that would be awesome !

**Pull request merged**